app/src/main/java/org/sopt/carrot/data/service/ProductApiService.kt

```diff
@@ -13,7 +13,7 @@ interface ProductApiService {
     @GET("/api/search")
     suspend fun searchProducts(
         @Query("keyword") keyword: String
-    ): Response<SearchResponse>
+    ): Response<BaseResponse<ResponseSearchDto>>
```

Comment:
p1: BaseResponse가 정의되어 있어서 추가적인 Response 사용은 제거해줘도 될 거 같습니다!!(+ 추가로 api를 명시하는 것도 좋은방법이지만, 현재 프로젝트 기준 `ProductApiService` 네이밍에서 Api 제거하는 것이 다른 service와의 일관성을 위해 좋을거 같습니다!
```suggestion
    ): BaseResponse<ResponseSearchDto>
```